proxy/src/telemetry/metrics/mod.rs

```diff
@@ -583,46 +561,36 @@ impl Aggregate {
-                            *metrics.tcp_connect_open_total(&labels).incr();
-                        },
-                    }
+                    *metrics.tcp().open_total(&labels).incr();
+                    *metrics.tcp().open_connections(&labels).incr();
                 })
             },
 
             Event::TransportClose(ref ctx, ref close) => {
-                let labels = Arc::new(TransportCloseLabels::new(ctx, close));
+                let labels = Arc::new(TransportLabels::new(ctx));
+                let close_labels = Arc::new(TransportCloseLabels::new(ctx, close));
                 self.update(|metrics| {
-                    *metrics.tcp_connection_duration(&labels) += close.duration;
-                    *metrics.sent_bytes(&labels) += close.tx_bytes as u64;
-                    *metrics.received_bytes(&labels) += close.tx_bytes as u64;
-                    match ctx.as_ref() {
-                        &ctx::transport::Ctx::Server(_) => {
-                            *metrics.tcp_accept_close_total(&labels).incr();
-                            // We don't use the accessor method here like we do
-                            // for all the other metrics so that we can just
-                            // use the transport open labels in `labels`
-                            // without having to ref-count it separately. Since
-                            // we're handling a close event, we expect those
-                            // labels to already be in the map from when the
-                            // transport was opened.
-                            *metrics.tcp_connections_open.values
-                                .get_mut(&labels.transport)
-                                .expect(
-                                    "observed TransportClose event for a \
-                                     transport that was not counted"
-                                )
-                                .decr();
-                        },
-                        &ctx::transport::Ctx::Client(_) => {
-                            *metrics.tcp_connect_close_total(&labels).incr();
-                        },
-                    };
+                    *metrics.tcp().write_bytes_total(&labels) += close.tx_bytes as u64;
+                    *metrics.tcp().read_bytes_total(&labels) += close.rx_bytes as u64;
+
+                    *metrics.tcp().connection_duration(&close_labels) += close.duration;
+                    *metrics.tcp().close_total(&close_labels).incr();
+
+                    // We don't use the accessor method here like we do for all the other
+                    // metrics so that we can just use the transport open labels in
+                    // `labels` without having to ref-count it separately. Since we're
+                    // handling a close event, we expect those labels to already be in the
+                    // map from when the transport was opened.
```

Comment:
I don't think this comment is still accurate, as we _do_ now construct and ref-count a separate set of `TransportLabels` --- it seems like the main reason we no longer use the accessor method is so that we can print the error message rather than creating a new gauge with the value 0 and decrementing it?